extensions/objects/templates/ImageWithRegionsEditor.js

```diff
@@ -272,15 +622,83 @@ oppia.directive('imageWithRegionsEditor', [
         $scope.onMouseoverRegion = function(index) {
           if ($scope.hoveredRegion === null) {
             $scope.hoveredRegion = index;
+            $scope.moveRegion = index;
+          }
+        };
+        $scope.onMouseMoveRegion = function(index) {
```

Comment:
What if you just didn't run this function while the user is dragging? That way you wouldn't need to worry about your "flag" variable. And plus the fact that onMouseMoveRegion and resizeRegion are both being constantly called while dragging seems unsettling.